x/oracle/client/common/proof/proof.go

```diff
@@ -80,6 +85,25 @@ func (o *OracleDataProof) encodeToEthData(blockHeight uint64) ([]byte, error) {
 	)
 }
 
+type RequestsCountProof struct {
```

Comment:
I think it's time to split each endpoint into multiple files and `single_result.go`, `multi_result.go` and `request_count.go` to pack data structure and handler code in the same file and create another file that contains shared code.